### DISCUSSION TITLE
flux bootstrap wont reconcile

### DISCUSSION THREAD

**Discussion by drizzef** [fluxcd/flux2#3461] [General] [answered]
Hi, 
i created a repo with the following structure(similar to [this](https://github.com/fluxcd/flux2-kustomize-helm-example)):

```
├── apps
│   ├── base
│   ├── production 
└── clusters
    ├── production
        ├── flux-system
        ├── apps.yaml
```

I ran the following code to bootstrap:
```
flux --kubeconfig ~/.kube/prod bootstrap git \    
  --url=https://repo.url \
  --branch=master \
  --password=**** \
  --token-auth=true --path=clusters/production
```

and the following error is displaying on the flux error logs:
```
kustomize build failed: accumulating resources: accumulation err='accumulating resources from './flux-system': read /tmp/kustomization-906255614/clusters/production/flux-system: is a directory': recursed merging from path '/tmp/kustomization-906255614/clusters/production/flux-system': may not add resource with an already registered id: Kustomization.v1beta2.kustomize.toolkit.fluxcd.io/flux-system.flux-system
```

And the apps.yaml which contains all the CRDs such as the kustomized app under `apps/production` folder won't applied to the cluster automatically. 

What am i missing?

Thanks!

**Comment by kingdonb** [comment, marked as answer]
Do you have a `flux-system` Kustomization in `apps.yaml`?

The error indicates that there's a duplicate definition for `flux-system` in the `flux-system` namespace. Given the filesystem tree you showed, that's my best guess at where you'll find it. Please make sure you have done `git pull` as well so that you are definitely looking at the latest version of the manifests.

If there are any other definitions of `flux-system` in the path that you apply, Kustomize controller will fail and the bootstrap will fail.

**Reply by drizzef** [reply]
you are right! thank you very much! i dont understand how i missed it
